packages/@react-spectrum/s2/src/ComboBox.tsx

```diff
@@ -383,11 +431,331 @@ export function ComboBoxSection<T extends object>(props: ComboBoxSectionProps<T>
+  if (size === 'S') {
+    menuOffset = 6;
+  } else if (size === 'M') {
+    menuOffset = 6;
+  } else if (size === 'L') {
+    menuOffset = 7;
+  } else {
+    menuOffset = 8;
+  }
+
+  let triggerRef = useRef<HTMLDivElement>(null);
+    // Make menu width match input + button
+  let [triggerWidth, setTriggerWidth] = useState<string | null>(null);
+  let onResize = useCallback(() => {
+    if (triggerRef.current) {
+      let inputRect = triggerRef.current.getBoundingClientRect();
+      let minX = inputRect.left;
+      let maxX = inputRect.right;
+      setTriggerWidth((maxX - minX) + 'px');
+    }
+  }, [triggerRef, setTriggerWidth]);
+
+  useResizeObserver({
+    ref: triggerRef,
+    onResize: onResize
+  });
+
+  let state = useContext(ComboBoxStateContext);
+  let timeout = useRef<ReturnType<typeof setTimeout> | null>(null);
+  let [showLoading, setShowLoading] = useState(false);
+  let isLoadingOrFiltering = loadingState === 'loading' || loadingState === 'filtering';
+  {/* Logic copied from S1 */}
+  let showFieldSpinner = useMemo(() => showLoading && (isOpen || menuTrigger === 'manual' || loadingState === 'loading'), [showLoading, isOpen, menuTrigger, loadingState]);
+  let spinnerId = useSlotId([showFieldSpinner]);
+
+  let inputValue = state?.inputValue;
+  let lastInputValue = useRef(inputValue);
+  useEffect(() => {
+    if (isLoadingOrFiltering && !showLoading) {
+      if (timeout.current === null) {
+        timeout.current = setTimeout(() => {
+          setShowLoading(true);
+        }, 500);
+      }
+
+      // If user is typing, clear the timer and restart since it is a new request
+      if (inputValue !== lastInputValue.current) {
+        clearTimeout(timeout.current);
+        timeout.current = setTimeout(() => {
+          setShowLoading(true);
+        }, 500);
+      }
+    } else if (!isLoadingOrFiltering) {
+      // If loading is no longer happening, clear any timers and hide the loading circle
+      setShowLoading(false);
+      if (timeout.current) {
+        clearTimeout(timeout.current);
+      }
+      timeout.current = null;
+    }
+
+    lastInputValue.current = inputValue;
+  }, [isLoadingOrFiltering, showLoading, inputValue]);
+
+  useEffect(() => {
+    return () => {
+      if (timeout.current) {
+        clearTimeout(timeout.current);
+      }
+      timeout.current = null;
+    };
+  }, []);
+
+  let renderer;
+  let listBoxLoadingCircle = (
+    <UNSTABLE_ListBoxLoadingSentinel
+      // TODO: we only want the listbox loading sentinel row to have a height when performing a loadingMore
+      // Unlike table, this works because the listbox's initial load isn't triggerd by the sentintel and thus the first
+      // observation occurs when we've already loaded our first set of items rather than starting from empty, therefor flipping
+      // isLoading here from true to false and triggering the creation of a new InterserctionObserver
+      isLoading={loadingState === 'loadingMore'}
```

Comment:
An interesting facet about how useLoadMoreSentinel works. Prior to some additional changes in layout, virtualized TableView ran into issues triggering additional `loadMore` calls when it was passing `isLoading={loadingState === 'loadingMore'}` to its own sentinel. This is because the sentinel never got a prop change before and after the initial load finished because the `loadingState` at that point was `loading` and thus we never caused a second `shouldLoadMore` check to happen in `useLoadMoreSentinel`